### PR TITLE
Node api generation

### PR DESCRIPTION
**Pull request by binarymist**
- [x] Provided backwards compatibility for those wishing to continue using callbacks, while those wishing to take advantage of promises can now do so
  - [x] The apiKey is now only required on the initial ZapClient instantiation

Addresses #5044

### CODE REVIEW

**Comment by binarymist** [comment]
Switching to tag 2.7.0 as suggested in https://github.com/zaproxy/zap-api-nodejs/pull/1#issuecomment-433961996 yields:

**Comment by thc202** [comment]
Are all the libraries under the lib directory in the module's dependencies?

**Comment by binarymist** [comment]
I haven't removed any

**Comment by thc202** [comment]
Right, but you might need to add.

File: src/org/zaproxy/zap/extension/api/NodeJSAPIGenerator.java [hunk 1]
```diff
@@ -31,12 +32,17 @@
 
 public class NodeJSAPIGenerator extends AbstractAPIGenerator {
     
+    /**
+	 * Default output directory in zap-api-nodejs project.
+	 */
+    private static final String DEFAULT_OUTPUT_DIR = "../zap-api-nodejs/src/zapv2/";
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/api/NodeJSAPIGenerator.java]: Is the path correct? In zaproxy/zap-api-nodejs#1 the files are being added directly to `src` dir (i.e. not to `src/zapv2`).

Comment (binarymist) [reply src/org/zaproxy/zap/extension/api/NodeJSAPIGenerator.java]: Yes, I didn't see any reason for a zapv2 directory, that's what versioning is for, unless there is something I've missed?

Comment (thc202) [reply src/org/zaproxy/zap/extension/api/NodeJSAPIGenerator.java]: I'm good with that, we should drop the `zapv2` from this path then.

Comment (binarymist) [reply src/org/zaproxy/zap/extension/api/NodeJSAPIGenerator.java]: Just realised I missed this, thought I'd got it, but obviously hadn't
fe8074c

**Comment by thc202** [comment]
Thanks!

**Pull request merged**